File: TFSWay.DomainModel/Entities/AuthorisedPerson.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace TFSWay.DomainModel.Entities
{
    public class AuthorisedPerson
    {
        [Key]
        public int AuthorisedPersonId { get; set; }
        public int? CompanyId { get; set; }
        public string Name { get; set; }
        public string Role { get; set; }
        public string ContactNumber { get; set; }
        public string LandlineNumber { get; set; }
        public string Email { get; set; }
        public int? AuthorisedPersonAgencyId { get; set; }
        public int? AuthorisedPersonPromoterId { get; set; }
        public string IsCoordinator { get; set; }
        public DateTime? CreatedDate { get; set; }
        public string CreatedBy { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public string UpdatedBy { get; set; }
    }
}

File: TFSWayUI/src/app/authorizedperson/authorisedperson.ts
export interface IAuthorisedPerson {

    authorisedPersonId: number;
    companyId?: number
    authorisedPersonPromoterId?: number;
    authorisedPersonAgencyId?: number;
    agencyType: string;
    agencyName: string
    agencyAddress: string 
    name: string;
    role: string;
    contactNumber: string;
    landlineNumber: string;
    email: string;
    chkCoordinator?: boolean;
    isCoordinator: string;
    createdDate: string;
    createdBy: string;
    updatedDate: string;
    updatedBy: string;
}

export enum AgencyType {
    Consultant = 1,
    Company = 2,
    Lender = 3,
    LE = 4,
    LFA = 5,
    LIA = 6,
    LLC = 7,
    Promoter = 8,
 }
